lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp

```diff
@@ -121,6 +121,46 @@ class RecomposeSelectFill_ : public OpRewritePattern<AtenFill_TensorOp> {
     return success();
   }
 };
+
+class RecomposeSplitTensorGetItemOp
+    : public OpRewritePattern<Aten__Getitem__TOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(Aten__Getitem__TOp op,
+                                PatternRewriter &rewriter) const override {
+    // recompose AtenSplitTensorOp + __getitem__t to AtenSliceTensorOp
+    auto splitTensorOp =
+        dyn_cast<AtenSplitTensorOp>(op.getList().getDefiningOp());
+    if (!splitTensorOp)
+      return failure();
+    if (isListPotentiallyMutated(op.getList()))
```

Comment:
Same comment here as in your other PR: https://github.com/llvm/torch-mlir/pull/2058#discussion_r1192694123